src/components/Map/LeafletDrawTools.js

```diff
@@ -206,6 +209,35 @@ export default function LeafletDrawTools(props) {
     dispatch(uploadedShapeFileGeoJSON(null));
   }, [shapeFileGeoJSON, dispatch]);
 
+  // Watches for an area to be added from search Places and adds its layer to the draw group ref
+  useEffect(() => {
+    if (!searchPlacesGeoJSON) {
+      return;
+    }
+    const layer = L.geoJSON(searchPlacesGeoJSON);
+    leafletDrawFeatureGroupRef.current.addLayer(layer);
+    const areaName = `Area ${areaNumber}`;
+    const bufferLayer = createBufferLayer(layer);
+    const geojson = enrichGeoJsonWithProperties(layer, bufferLayer, areaName);
+    setAreaNumber(areaNumber + 1);
+    layer.bindTooltip(areaName, { direction: 'center', permanent: true, className: classes.leafletTooltips });
+    const dummyFeatureGroup = L.featureGroup();
+    dummyFeatureGroup.addLayer(layer);
+    const dummyFeatureGroupGeoJSON = dummyFeatureGroup.toGeoJSON();
+    const zonalStatsPromise = zonalStatsAPI(dummyFeatureGroupGeoJSON, selectedRegion);
+    zonalStatsPromise.then((data) => {
+      dispatch(setEmptyState(false));
+      data.features.forEach((feature) => {
+        geojson.properties.zonalStatsData = feature.properties.mean; // Should only be 1 feature
+        dispatch(addNewFeatureToDrawnLayers(geojson));
+        setDrawAreaDisabled(false);
+      });
+    });
+
+    dispatch(addSearchPlacesGeoJSON(null));
```

Comment:
I see why you're doing this but I'm not sure if it's necessary? It seems like it will call your use effect again which will just return because it's null but it's extra bandwidth that may not be needed.